src/main/java/DukeSaver.java

```diff
@@ -0,0 +1,76 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * Encapsulates the saving system for tasks.
+ */
+public class DukeSaver {
+
+    /**
+     * Holds the path to the saved data file.
+     */
+    private String savePath;
+
+    /**
+     * Constructor for the save system.
+     *
+     * @param savePath Path to the save file.
+     */
+    public DukeSaver(String savePath) {
+        this.savePath = savePath;
+        initSaveDir();
+    }
+
+    /**
+     * Initializes the directory and save file.
+     * If already exists, does not do anything.
+     */
+    public void initSaveDir() {
+        try {
+            File saveFile = new File(this.savePath);
+            saveFile.getParentFile().mkdirs();
+            saveFile.createNewFile();
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+    }
```

Comment:
Have you considered making this method private? Seems like it is already invoked in the constructor and the client likely won't need it